onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -121,21 +123,73 @@ NnapiExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view
 
   std::unordered_set<std::string> node_outputs_in_current_group{};
 
+  std::vector<std::unique_ptr<INodeUnit>> node_unit_holder;
+  std::unordered_map<const Node*, INodeUnit*> node_unit_map;
+
+  // TODO share this code
+  // TODO make this similar to qdq transformer
```

Comment:
The QDQ approach for EPs needs to be easily re-usable, so we may need to split up what it does into 1) identifying all QDQ node groups, 2) adding EP specific logic to select which of those can be handled. 

In Compile you just need to re-run step 1 (unless we add something to pass state between GetCapability and Compile), as any nodes that Compile is being called with, that's in a QDQ node group, is supported. 

Additional EP specific selection logic could be via derived classes from the basic selectors - probably what the CPU EP setup would do. In the NNAPI case we possibly don't need any derived classes if the op support checker is going to provide the additional logic.

Reply:
Yes, we can add shared QDQ select logic (line 126~165 here), which at least can be used in both GetCapability and Compile